### ISSUE TITLE
need a safe way to truncate local datastore

### ISSUE THREAD

**Issue by jbenet** [ipfs/kubo#245]
`rm -rf $IPFS_DIR/datastore` will break things

**Comment by whyrusleeping** [comment]
this would still be nice to have:
```
$ ipfs repo reset
Are you sure you want to delete all of your ipfs data? [y/N]
```

**Comment by whyrusleeping** [comment]
this could essentially just be:
```
for key in node.Blockstore.KeysChan() {
    node.Blockstore.Delete(key)
}
```

**Comment by whyrusleeping** [comment]
Revision:

- remove all pins on all data
- iterate all blocks and remove them

**Comment by rascalking** [comment]
Hi, I'm trying to pick this up for hacktoberfest, but I cannot for the life of me figure out how to get the confirmation prompt to fire on the client side.  Anyone have any pointers?

**Comment by schomatis** [comment]
> The "just delete every block" method works fine for the `flatfs` datastore but will have terrible performance on badger.

Iterating doesn't seem that expensive in Badger since we're basically doing a serial read of big files (the most expensive part would be decoding the headers of each value block) but don't quote me on this, I should run a few test to be sure.

Out of ignorance, we can't just delete the Badger DB because it's storing more than just the files that the user added, right? Some metadata?

**Comment by Stebalien** [comment]
> Out of ignorance, we can't just delete the Badger DB because it's storing more than just the files that the user added, right? Some metadata?

Yes. We'd have to either just delete the blocks or migrate everything but the blocks to a new datastore and then delete the old one.

> Iterating doesn't seem that expensive in Badge

IIRC, badger has behaves really poorly when deleting a ton of data. @magik6k?

**Comment by schomatis** [comment]
> IIRC, badger has behaves really poorly when deleting a ton of data. @magik6k?

Actually you're right, deleting itself is not the most expensive part (although you'll be creating new delete records for every value stored) but purging the db after deleting everything (which implies a search for every delete key and many compact operations) is very cpu intensive.
